src/avro/src/reader.rs

```diff
@@ -750,7 +747,16 @@ impl<'a> SchemaResolver<'a> {
                         self.reader_to_resolved_names.insert(r_index, resolved_idx);
                         self.named.push(None);
                         let piece =
-                            self.resolve_named(writer.root, reader.root, w_index, r_index)?;
+                            match self.resolve_named(writer.root, reader.root, w_index, r_index) {
+                                Ok(piece) => piece,
+                                Err(e) => {
+                                    // This could have just been a "?", except that we need to clean up
+                                    // the placeholder values that were added above.
```

Comment:
micronit TIOLI: this could just be `// clean the placeholder values that were added above`